src/librustc_lint/unused.rs

```diff
@@ -146,6 +146,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
 
         let t = cx.tables.expr_ty(&expr);
         let ty_warned = match t.sty {
+            ty::TyTuple(ref tys, _) if tys.is_empty() => return,
+            ty::TyNever => return,
```

Comment:
Should this also apply to `enum Void {}`, and other such uninhabited types?  (Possibly like https://github.com/rust-lang/rust/pull/38069?)